coding-the-legislation/10_basic_example.md

```diff
@@ -17,19 +17,30 @@ class flat_tax_on_salary(Variable):
 ```
 
 Let's explain in details the different parts of the code:
-- `class flat_tax_on_salary(Variable):` declares a new variable with the name `flat_tax_on_salary`.
-- Metadatas:
-  - `value_type = float` declares the type of the variable. Possible types are the basic python types:
+
+### The name of the variable
+
+`class flat_tax_on_salary(Variable):` declares a new variable with the name `flat_tax_on_salary` in accordance with our [naming conventions](../contribute/variables-naming.md).
+
+### The metadata
+
+All variables have a set of metadata.
+* `value_type` defines the type of the formula output. Possible types are the basic python types,
+but note that they are in fact [`numpy.array`](./coding-the-legislation/25_vectorial_computing.md).
+Available types are :
     - `bool`: boolean
     - `date`: date
     - `Enum`: discrete value (from an enumerable). [See details](20_input_variables.md#advanced-example-enumerations-enum) in the next section.
-    - `float`: float
+    - `float`: float32 (for memory usage constraints)
```

Comment:
`float32` is not a Python builtin. Maybe clarify this with:

- `float`: float (Note that to reduce memory usage, float are stored on 32 bits using NumPy's `float32`)